frontend/src/deviceDetailsPage/__tests__/DeviceLogTableWrapper.test.jsx

```diff
@@ -0,0 +1,102 @@
+import React from "react";
+import Enzyme from "enzyme";
+import Adapter from "enzyme-adapter-react-16";
+import { describe, expect, it, jest } from "@jest/globals";
+
+import DeviceLogTableWrapper from "../DeviceLogTableWrapper";
+import LogInfo from "../../model/LogInfo";
+import LogsTable from "../../loglist/LogsTable";
+import DeviceInfo from "../../model/DeviceInfo";
+
+import * as LogApi from "../../api/LogApi";
+
+Enzyme.configure({ adapter: new Adapter() });
+jest.mock("../../api/LogApi");
+
+describe("<DeviceLogTableWrapper/> Class Component", () => {
+  let wrapper;
+  const dummyDevice = new DeviceInfo(
+    "serial",
+    "lastCommunication",
+    "publicIpAddress",
+    "privateIpAddress",
+    "displayName",
+    "status",
+    "channels",
+    "deviceType",
+    "extras"
+  );
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("handleLogsChange()", () => {
+    const expectedLogs = [new LogInfo(5)];
+
+    LogApi.getDeviceLogs.mockReturnValue(Promise.resolve(expectedLogs));
+    wrapper = Enzyme.shallow(<DeviceLogTableWrapper device={dummyDevice} />);
```

Comment:
just checking that i understand correctly - the `expectedLogs` and call to `LogApi` are needed so you can do the `shallow`?

Reply:
Yep. Because logApi is mocked in this file & is called in componentDidMount(), which happens on render, even if only shallow. We need to fake the implementation. If I just leave it blank, it complains that the value is undefined.